src/hooks/useForumConnect.ts

```diff
@@ -1,49 +1,40 @@
-import { useCallback, useEffect, useState } from 'react'
+import { useCallback } from 'react'
 
 import { Web3Provider } from '@ethersproject/providers'
-import useAuthContext from 'decentraland-gatsby/dist/context/Auth/useAuthContext'
-import useTrackContext from 'decentraland-gatsby/dist/context/Track/useTrackContext'
 
 import { Governance } from '../clients/Governance'
-import { AccountType } from '../components/Modal/IdentityConnectModal/AccountsConnectModal'
 import { SegmentEvent } from '../entities/Events/types'
-import { GATSBY_DISCOURSE_CONNECT_THREAD, MESSAGE_TIMEOUT_TIME } from '../entities/User/constants'
+import { GATSBY_DISCOURSE_CONNECT_THREAD } from '../entities/User/constants'
+import { Account, AccountType } from '../entities/User/types'
 import { DISCOURSE_API } from '../entities/User/utils'
 import { openUrl } from '../helpers'
-import Time from '../utils/date/Time'
 
-import useClipboardCopy from './useClipboardCopy'
-import useTimer from './useTimer'
+import useValidationSetup from './useValidationSetup'
 
 export const THREAD_URL = `${DISCOURSE_API}${
   DISCOURSE_API.endsWith('/') ? '' : '/'
 }t/${GATSBY_DISCOURSE_CONNECT_THREAD}/10000`
 const VALIDATION_CHECK_INTERVAL = 10 * 1000 // 10 seconds
 
-const getMessage = async () => Governance.get().getValidationMessage()
+const getMessage = async () => Governance.get().getValidationMessage(Account.Forum)
 
 export default function useForumConnect() {
-  const [user, userState] = useAuthContext()
-  const track = useTrackContext()
-
-  const [clipboardMessage, setClipboardMessage] = useState('')
-  const { handleCopy } = useClipboardCopy(Time.Second)
-  const { startTimer, resetTimer, time } = useTimer(MESSAGE_TIMEOUT_TIME / 1000 - 1)
-  const [validatingProfile, setValidatingProfile] = useState<NodeJS.Timer>()
-  const [isValidated, setIsValidated] = useState<boolean>()
-
-  useEffect(() => {
-    if (time <= 0 && validatingProfile) {
-      clearInterval(validatingProfile)
-      setValidatingProfile(undefined)
-    }
-  }, [time, validatingProfile])
-
-  useEffect(() => {
-    if (isValidated !== undefined) {
-      resetTimer()
-    }
-  }, [isValidated, resetTimer])
+  const {
+    user,
+    userState,
+    track,
+    clipboardMessage,
+    setClipboardMessage,
+    handleCopy,
+    startTimer,
+    resetTimer,
+    time,
+    validatingProfile,
+    setValidatingProfile,
+    isValidated,
+    setIsValidated,
+    resetValidation,
+  } = useValidationSetup()
 
   const getSignedMessage = useCallback(async () => {
```

Comment:
Couldn't this function be part of the useValidationSetup hook? And conditionally fetch the message according an argument passed to useValidationSetup. E.g: `useValidationSetup(Account.Forum)`